Library/Homebrew/extend/git_repo_path.rb

```diff
@@ -7,12 +7,14 @@
 # Extensions to {Pathname} for querying Git repository information.
 # @see Utils::Git
 # @api private
-module GitRepositoryExtension
+class GitRepoPath < SimpleDelegator
   extend T::Sig
 
+  alias pathname __getobj__
+
   sig { returns(T::Boolean) }
-  def git?
-    join(".git").exist?
+  def git_repo?
```

Comment:
`GitRepoPath.new(repo).git_repo?` is repetitive, yes, but the problem is in `Library/Homebrew/tap.rb`. I think some minor redundancy elsewhere is preferable to the below, wdyt?

```diff
diff --git a/Library/Homebrew/tap.rb b/Library/Homebrew/tap.rb
index d1e9c4850..4c17062f6 100644
--- a/Library/Homebrew/tap.rb
+++ b/Library/Homebrew/tap.rb
@@ -135,7 +135,7 @@ class Tap
   def remote
     return default_remote unless installed?

-    @remote ||= path.git_origin
+    @remote ||= path.origin
   end

   # The remote repository name of this {Tap}.
@@ -163,28 +163,28 @@ class Tap

   # True if this {Tap} is a Git repository.
   def git?
-    path.git_repo?
+    path.repo?
   end

   # git branch for this {Tap}.
   def git_branch
     raise TapUnavailableError, name unless installed?

-    path.git_branch
+    path.branch
   end

   # git HEAD for this {Tap}.
   def git_head
     raise TapUnavailableError, name unless installed?

-    @git_head ||= path.git_head
+    @git_head ||= path.head
   end

   # Time since last git commit for this {Tap}.
   def git_last_commit
     raise TapUnavailableError, name unless installed?

-    path.git_last_commit
+    path.last_commit
   end

   # The issues URL of this {Tap}.
@@ -385,20 +385,20 @@ class Tap
       $stderr.ohai "#{name}: changed remote from #{remote} to #{requested_remote}" unless quiet
     end

-    current_upstream_head = path.git_origin_branch
-    return if requested_remote.blank? && path.git_origin_has_branch?(current_upstream_head)
+    current_upstream_head = path.origin_branch
+    return if requested_remote.blank? && path.origin_has_branch?(current_upstream_head)

     args = %w[fetch]
     args << "--quiet" if quiet
     args << "origin"
     safe_system "git", "-C", path, *args
-    path.git_origin_set_head_auto
+    path.origin_set_head_auto

-    new_upstream_head = path.git_origin_branch
+    new_upstream_head = path.origin_branch
     return if new_upstream_head == current_upstream_head

-    path.git_rename_branch old: current_upstream_head, new: new_upstream_head
-    path.git_branch_set_upstream local: new_upstream_head, origin: new_upstream_head
+    path.rename_branch old: current_upstream_head, new: new_upstream_head
+    path.branch_set_upstream local: new_upstream_head, origin: new_upstream_head

     return if quiet
```

Reply:
Re-reading the above: yeh, I agree moreso with this that it's confusing that `path` doesn't return a `Pathname`. Feels like this method should have `git_` somewhere in it as it's now returning a `GitRepoPath`